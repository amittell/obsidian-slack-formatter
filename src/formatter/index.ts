/**
 * Slack Formatter exports
 */
export { SlackFormatter } from './slack-formatter';

// Processors
export * from './processors/base-processor';
export * from './processors/username-processor';
export * from './processors/url-processor';
export * from './processors/code-block-processor';
export * from './processors/emoji-processor';
export * from './processors/thread-link-processor';
export * from './processors/unified-processor';
export * from './processors/attachment-processor';

// Stages
export * from './stages/flexible-message-parser';
export * from './stages/improved-format-detector';
export * from './stages/preprocessor';
export * from './stages/postprocessor';
export * from './stages/format-detector';
export * from './stages/message-parser';

// Strategies
export * from './strategies/base-format-strategy';
export * from './strategies/standard-format-strategy';
export * from './strategies/bracket-format-strategy';
export * from './strategies/mixed-format-strategy';
export * from './strategies/format-strategy-factory';
