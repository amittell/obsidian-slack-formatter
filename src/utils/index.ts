/**
 * Central export point for all utility functions
 */

export * from './datetime-utils';
export * from './emoji-utils';
export * from './text-utils';
export * from './username-utils';
export * from './json-utils'; // Add export for json-utils
export * from './constants';
export * from './validation-utils';
export * from './text-encoding-utils';
export * from './text-normalization-engine';
export * from './content-preservation-validator';
export * from './content-sanitization-pipeline';
