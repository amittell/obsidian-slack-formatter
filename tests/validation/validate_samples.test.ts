import fs from 'fs';
import path from 'path';
import { SlackFormatter } from '../../src/formatter/slack-formatter'; // Relative path from tests/validation
import { DEFAULT_SETTINGS } from '../../src/settings'; // Relative path
import { parseJsonMap } from '../../src/utils';
import { TestLogger } from '../helpers'; // Relative path

// Determine project root assuming tests run from the project root directory
const projectRoot = process.cwd();
const samplesDir = path.resolve(projectRoot, 'samples');

// Get all .txt files from the samples directory
let sampleFiles: string[] = [];
try {
  sampleFiles = fs.readdirSync(samplesDir).filter(file => file.endsWith('.txt'));
  TestLogger.log(`Found sample files: ${sampleFiles.join(', ')}`);
} catch (err) {
  TestLogger.error(`Error reading samples directory at ${samplesDir}:`, err);
  // Optionally, throw the error or exit if samples are critical
  // throw new Error(`Could not read samples directory: ${samplesDir}`);
}

// Initialize formatter (using default settings)
const userMap = parseJsonMap(DEFAULT_SETTINGS.userMapJson || '{}', 'User Map') ?? {};
const emojiMap = parseJsonMap(DEFAULT_SETTINGS.emojiMapJson || '{}', 'Emoji Map') ?? {};
// Ensure settings are passed correctly, potentially cloning to avoid modification issues
const settings = { ...DEFAULT_SETTINGS };
const formatter = new SlackFormatter(settings, userMap, emojiMap);

// Describe block for Jest
describe('Slack Formatter Validation Samples', () => {
  if (sampleFiles.length === 0) {
    it('should handle missing samples directory gracefully', () => {
      TestLogger.warn(`No sample files found in ${samplesDir}. Skipping validation tests.`);
      expect(true).toBe(true); // This is a valid scenario - no samples directory
    });
    return;
  }

  sampleFiles.forEach(sampleFile => {
    test(`should correctly format ${sampleFile}`, () => {
      const samplePath = path.join(samplesDir, sampleFile);
      let sampleContent = '';
      try {
        sampleContent = fs.readFileSync(samplePath, 'utf-8');
      } catch (err) {
        throw new Error(`Could not read sample file: ${samplePath}. Error: ${err}`);
      }

      // Format the content
      let formattedOutput = '';
      try {
        // Ensure the formatter is re-initialized or state is reset if necessary between tests,
        // although for this stateless formatter, it might not be strictly needed.
        // Re-parsing maps just in case, though defaults shouldn't change.
        const currentSettings = { ...DEFAULT_SETTINGS };
        const currentUserMap = parseJsonMap(currentSettings.userMapJson || '{}', 'User Map') ?? {};
        const currentEmojiMap =
          parseJsonMap(currentSettings.emojiMapJson || '{}', 'Emoji Map') ?? {};
        const currentFormatter = new SlackFormatter(
          currentSettings,
          currentUserMap,
          currentEmojiMap
        );
        formattedOutput = currentFormatter.formatSlackContent(sampleContent);
      } catch (err) {
        TestLogger.error(`Error formatting ${sampleFile}:`, err);
        // Optionally re-throw or fail the test differently
        // We'll let the snapshot failure indicate the error for now.
        formattedOutput = `FORMATTING ERROR for ${sampleFile}: ${err}`;
      }

      // Compare with snapshot named after the sample file
      // Jest automatically creates/updates .snap files in a __snapshots__ directory
      expect(formattedOutput).toMatchSnapshot();
    });
  });
});
