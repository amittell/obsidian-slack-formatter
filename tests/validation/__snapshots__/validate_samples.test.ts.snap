// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack Formatter Validation Samples should correctly format duckcreek-sample.txt 1`] = `
"> [!slack]+ Message from User1
> **Time:** 3:26 PM
>
> was added to guidewire-gtm-pursuit by User2
> .

> [!slack]+ Message from User3
> **Time:** 3:31 PM
>
> Hey
> [[user4]]
> - as an aside, just visited Duck Creek's payments page again. They've now fully baked in emburse as a payments orchestrator - and are calling it Duck Creek Payments.. Whats interesting is that they haven't included our logo (we were in from an acquiring pov) but have included Adyen. Given their book - i find this a touch concerning. I know we can't pursue two strategic partnerships - and that its not the best use of Stripe to be an endpoint in another orchestrator. But considering they are the #2 player, and possibly still #1 on pay-ins, I think we should engage a plats AE to see if we can at least get our payments solution into the platform - as much from a "no disqualification" pov as anything else.
> Link: https://www.duckcreek.com/wp-content/uploads/2024/1/Duck-Creek-Payments-Solutions.pdf
> Current users: https://www.duckcreek.com/customer/
> C
> [[user5]]
> ,
> [[user6]]
> (edited)

> [!slack]+ Message from User7
> **Time:** 3:43 PM
>
> was added to guidewire-gtm-pursuit by User5
> .

> [!slack]+ Message from User8
> **Time:** 2:48 PM
>
> hi team, just noticed this press release that mentions Stripe as a Guidewire partner:
> Guidewire PartnerConnect Ecosystem Surpasses 110 Cloud-Native Integrations, Enables Insurers to Accelerate Business Growth and Unlock New Opportunities
>
> :eyes-intensifies: 1"
`;

exports[`Slack Formatter Validation Samples should correctly format emoji-channel-sample.txt 1`] = `
"> [!slack]+ Message from User1
> **Time:** 3:04 PM
>
> replied to a thread:
> wtb i.e. cookie monster, but blurple.

> [!slack]+ Message from User2
> **Time:** 12:46 PM
>
> Anyone else got beef with emoji?
> Our partner merchant ids are in the format krngloballive:&lt;suffix&gt; so half of the time they get pasted as just "krngloballive"
> Also noticing the replacement a lot in automated messages since we've a bunch of ruby packages called partner

> [!slack]+ Message from User3
> **Time:** 2:23 PM
>
> test

> [!slack]+ Message from User4
> **Time:** 3:18 PM
>
> wtb (where bufo looks despondent and/or bored)

> [!slack]+ Message from User5
> **Time:** 3:24 PM
>
> wtb
>
> :bufo-coding: 5 :bufo-happy-coding: 2 :bufo-yearns-for-the-mines: 9

> [!slack]+ Message from User6
> **Time:** 1:52 PM
>
> wtb
>
> 2 months agoView thread
> This message was deleted.
>
> :cat-huhh: 3

> [!slack]+ Message from User3
> **Time:** 2:23 PM
>
> wts: is a very low quality emoji for how often it gets used (purely because of bufo/SEO). if someone creates a cleaner version, I'l gladly give up the emoji namespace to them. It is a &gt;p99 bufo according to go/emoji-scoreboard (edited)
>
> :til-dark: 4 :bufo-clap: 5 :five-nines: 4

> [!slack]+ Message from User7
> **Time:** 8:01 PM
>
> 8:04
> big bless
> Custom response

> [!slack]+ Message from User8
> **Time:** 11:38 AM
>
> is there any way to get a dump of all our emojis names, for better searching?

> [!slack]+ Message from User9
> **Time:** 1:58 AM
>
> Does anyone have the template for where I can replace the parrot?
>
> :bufo-worried-doctor: 1

> [!slack]+ Message from User10
> **Time:** 2:59 PM
>
> I need an emoji for
>
> :bufo-takes-hubble: 6

> [!slack]+ Message from User11
> **Time:** 1:07 PM
>
> Has anyone studied bufo's migration? as I understand it many ex-stripes spread bufo to OpenAI, and Notion... where else?
> X (formerly Twitter)X (formerly Twitter)
> Notion ([[NotionHQ]]) on X
> Last year we launched custom emojis in Notion
> But how do you add 1.5M custom emojis to a workspace without breaking everything? (91 kB)
>
> 9+
>
> :future-bufos: 4

> [!slack]+ Message from User12
> **Time:** 6:22 PM
>
> went to reference the anikan princess leia me and I was extremely disappointed we don't have a princess leia emoji... I thought we were nerds people
>
> :be-the-emoji-you-want-to-see: 1 :right-meme: 9 :r1-blurple: 1

> [!slack]+ Message from User13
> **Time:** 6:18 AM
>
> replied to a thread:
> Has anyone studied bufo's migration? as I understand it many ex-stripes spread bufo to OpenAI, and Notion... where else?
> does anyone know which bufo is this?
> image.png
>
> image.png
>
> View newer replies
>
> :bufo-ages-rapidly-in-the-void: 18

> [!slack]+ Message from User14
> **Time:** 12:11 PM
>
> :bufo寿限無寿限無五劫の擦り切れ海砂利水魚の水行末雲来末風来末食う寝る処に住む処やぶら小路の藪柑子パイポパイポパイポのシューリンガンシューリンガンのグーリンダイグーリンダイのポンポコピーのポンポコナ:
>
> :bufo-offers-eyes-but-the-provenance-is-disconcerting: 4 :blob-hug-broken-image: 2 :blob-hug-broken-image-r: 1 :bufo-waves-a-fish-around-because-this-smells-fishy-and-he-doesnt-want-his-message-to-be-misconstrued: 1

> [!slack]+ Message from User15
> **Time:** 1:58 PM
>
> replied to a thread:
> Has anyone studied bufo's migration? as I understand it many ex-stripes spread bufo to OpenAI, and Notion... where else?
> the github one has bufos that we don't have! https://github.com/knobiknows/all-the-bufo/blob/main/all-the-bufo/bufo-intern.png
> bufo-intern.png
> [knobiknows/all-the-bufo](https://github.com/knobiknows/all-the-bufo)knobiknows/all-the-bufo | Added by GitHub (23 kB)
>
> View newer replies

> [!slack]+ Message from User16
> **Time:** 3:59 PM
>
> Emojiteam can we remove this? For photosensitivity
>
> +1

> [!slack]+ Message from User17
> **Time:** 3:04 PM
>
> wtb / which is like   or but with pompoms (animation optional)
>
> 2 days agoView thread
>
> :bufo-cheerleader: 6

> [!slack]+ Message from User18
> **Time:** 1:28 PM
>
> is so cursed
>
> :yikes-harold: 5

> [!slack]+ Message from User19
> **Time:** 2:20 PM
>
> wtb to use when someone has linked to a private channel you're not in (see screenshot from Slack) (edited)

> [!slack]+ Message from User20
> **Time:** 3:43 PM
>
> Hi Team
> Is there a table in hubble where I can check who created a emoji? Any help would be appreciated :)

> [!slack]+ Message from User21
> **Time:** 8:28 PM
>
> wtb
>
> :big-yikes: 3 :bufo-blanket: 1 :fancy-bufo: 2 :bufo-eats-his-son-like-popcorn: 2 :bufo-what-have-you-done: 1 :wait-thats-illegal: 1

> [!slack]+ Message from User14
> **Time:** 6:57 PM
>
> From a fresh Slack instance. It's no way to live.
> Image from iOS
>
> :nice-1: 5 :nice-brat: 5 :nice-icon: 5 :nice-100: 5 :nice-colors: 6 :nice-nice: 5 :nice-blurple: 5 :haha-nice: 5 :small-nice: 5 :super-nice: 5 :nice-in-swedish: 5 :party-nice: 4 :double-nice: 4

> [!slack]+ Message from User22
> **Time:** 2:24 PM
>
> I'm looking for help removing a reactji, specifically looking to retire jira-archreview-ask since that forum is deprecated. I cleaned up the go/askbot-sheet to remove the behavior, but the emoji is owned by a former Stripe.
>
> 3 days agoView thread

> [!slack]+ Message from User19
> **Time:** 10:25 AM
>
> wtb ()
>
> Last reply 19 hours ago
>
> :double-ty: 7"
`;

exports[`Slack Formatter Validation Samples should correctly format fireside-sample.txt 1`] = `
"> [!slack]+ Message from User1
> **Time:** 12:00 PM
>
> Hi Stripes, here's the agenda for this week's Friday Fireside, with
> [[user2]]
> and
> [[user3]]
> We'l hear from Archive co-founder & CEO User4 and Co-founder & CTO User5. Archive powers profitable resale programs for apparel and footwear brands.
> [[user6]]
> and
> [[user7]]
> will update us on our six key financial metrics
> [[user8]]
> and
> [[user9]]
> from the Money Movement and Storage (MMS) team will demo key payouts features that improve merchant flexibility (and drive liquidity for Stripe)
> Patrick will take us through tabs!
>   Oh and if you haven't already: make sure to read the 2024 annual letter.
> See you there!

> [!slack]+ Message from User10
> **Time:** 11:51 AM
>
> channel is open and we'l get start in 10 minutes
>
> :bufo-reading: 3 :blob-omg: 4 :bufo-run: 1

> [!slack]+ Message from User11
> **Time:** 11:59 AM
>
> Rare blufo (blurfo?)
> ED: yeah, (edited)
>
> :blurfo-vibe: 9 :bufo-offers-blurfo: 1

> [!slack]+ Message from Benjamin Neuwirth
> **Time:** 12:02 PM
>
> So are these clothes that are returned? Or fully used and then resold?
>
> :good-question: 5

> [!slack]+ Message from Darren Jennings
> **Time:** 12:03 PM
>
> their icon is reminiscent of analogue.co

> [!slack]+ Message from Eoin Nugent
> **Time:** 12:03 PM
>
> there's a new shuppa startup! (...at least phonetically the same name) https://www.myshuppa.com/ groceries+alcohol in (at least) dublin
> Shuppa
> shuppa | Instant Grocery Delivery in Dublin
> Shuppa | Grocery delivery dublin | Wine delivery dublin | App to doorstep in minutes | Download the app & shop today!

> [!slack]+ Message from Miriam Keil
> **Time:** 12:08 PM
>
> Sounds like you didn't spend your youth thrifting clothes, P.

> [!slack]+ Message from Dan Nelson
> **Time:** 12:09 PM
>
> Let's force a reason for this to be on-chain... if I buy a used coat through Archive, I could see the previous owners and we could play the game of "who wore it best".
>
> 8 days agoView thread

> [!slack]+ Message from Anthony Voellm
> **Time:** 12:09 PM
>
> Selling my Tandy Coco2 for a premium!!!! True vintage 1980s (edited)
>
> Adam O'Donnell
>   12:09 PM
> This is a fascinating market.
>
> :equals-equals: 1

> [!slack]+ Message from User11
> **Time:** 12:10 PM
>
> economists: did someone say incentives?

> [!slack]+ Message from User1
> **Time:** 5:19 PM
>
> If you missed it, be sure to catch the replay available now on the new StripeTV (with playback speed, chapter markers, and transcript, oh my!)

> [!slack]+ Message from And
> **Time:** At Friday Fireside This week
>
> [[user3]]
> hosted and we heard from
> Archive co-founder & CEO User4 and Co-founder & CTO User5. Archive powers profitable resale programs for apparel and footwear brands
> [[user6]]
> and
> [[user7]]
> updated us on our six key financial metrics and the impact of/on GTM
> [[user8]]
> and
> [[user9]]
> from the Money Movement and Storage (MMS) team demo'd key payouts features that improve merchant flexibility (and drive liquidity for Stripe)
> Patrick took us through tabs! and then we heard from guest User12 on this week's QBR
> The replay, notes, and audio are up at go/f-stream, go/f-notes and go/f-audio.
>   Oh and if you haven't already: make sure to read the 2024 annual letter.
>
> :bufo-offers-stripetv: 2 :austin-spin: 1 :bufo-offers-stripetv-in-technicolor: 2"
`;

exports[`Slack Formatter Validation Samples should correctly format guidewire-sample.txt 1`] = `
"> [!slack]+ Message from Thread
> **Time:** 
>
> guidewire-gtm-pursuit

> [!slack]+ Message from User1
> **Time:** 10:39 AM
>
> We haven't notified them yet
> [[user2]]
> , we were hoping to get a more detailed BBG quote from User3 this week (still waiting to hear back).
> Estimated timeline Endava gave us was min-22weeks from when we kick off the engamement to build all of the integrations we scoped.
> One thing we would like to know
> [[user2]]
> is if Guidewire would be able to provide us with development licenses / instances for us to develop against? Endava are saying there is a $25k pass-through license cost from Guidewire to obtain this, which seems excessive and not very in the spirit of a strategic partnership with us.  (edited)
> Also sent to the channel

> [!slack]+ Message from User4
> **Time:** 8:19 PM
>
> [[user5]]
> organized a super positive Insurance Solutioning Workshop with Infosys in the NY office today .
> A more comprehensive update will be coming soon with additional AIs, but we wanted to get this one on your radar quickly
> [[user1]]
>
> [[user6]]
> since there's a 3/10 target date for a decision on the Guidewire Connector approach in the SteerCo doc:
> Infosys team showed high interest in our Guidewire Connector integration strategy-they're eager to hold a follow-up meeting to discuss partnering on building for the Marketplace. I know we're close with Endava, but the benefits Infosys mentioned make this worth exploring before gaveling on the build partner decision:
> They have 200+ Guidewire-certified employees.
> They already have sandbox instances we'd be able to use for the Connectors without paying to obtain them from Guidewire.
> They take on contracted liability for ongoing support/maintenance/upgrades of marketplace integrations.
> The majority of Insurance businesses they've worked with have started with pre-existing Guidewire marketplace integrations before moving on to custom builds. "Land with Guidewire then expand."
> They agreed to share a list of ~195 current/prospective Insurance users with business cases that Stripe may be able to solve for-we didn't ask about the overlap with Guidewire and can do so when the list is shared. Either way they agreed that driving customer leads our way, whether direct or through Guidewire, would be a key aspect of the partnership.
> Let me know if I've misrepresented or missed anything (and huge thanks to you and the rest of the team that contributed to the workshop
> [[user7]]
>
> [[user8]]
>
> [[user9]]
>
> [[user10]]
>
> [[user11]]
> ). (edited)

> [!slack]+ Message from User4
> **Time:** 8:26 PM
>
> c
> [[user12]]
>
> [[user2]]
>
> [[user13]]
>
> [[user14]]
> for viz/thoughts/concerns

> [!slack]+ Message from User1
> **Time:** 9:43 AM
>
> Good to have alternatives I think we should definitely at least give Infosys a shot at a proposal. Can you make an intro for me
> [[user4]]
> so we can get the ball rolling on it asap?

> [!slack]+ Message from User14
> **Time:** 9:44 AM
>
> That second bullet is highly insightful. Thank you

> [!slack]+ Message from User4
> **Time:** 12:15 PM
>
> Looped you into an intro
> [[user1]]
> !
>
> 🙏 1

> [!slack]+ Message from User6
> **Time:** 3:10 PM
>
> Hi all - I spoke to User1 about this earlier. We've already spent weeks working with Endava and Guidewire on this and I hesitate to bring in a new variable at this point. We are recommending Endava due to their deep experience with Stripe and the fact that they have already created the on-prem connector - we believe their ramp will be fast. User1 is working on our final recommendation to be shared XFN early next week. I think there is value in User1 reviewing our planned approach with Infosys to get their take and our A&C team engaging them in our GTM strategy but I don't recommend we ask them to pitch us a proposal at this point
>
> User14
>   Yesterday at 3:10 PM
> Agreed

> [!slack]+ Message from User14
> **Time:** 3:10 PM
>
> Agreed

> [!slack]+ Message from Agreed
> **Time:** 3:11 AM
>
> Get the endava stuff out there

> [!slack]+ Message from User7
> **Time:** 3:40 PM
>
> Thanks for the clarity
> [[user6]]
> but I think it may cause the wrong expectation with Infosys if we validate our approach with them if we've already decided to go with another partner. They were very eager to have a discussion around the potential of building a connector when
> [[user4]]
> proposed it. Shall we rethink having the call? I'd need to think about the way to frame it given they've been asking us to get in front of work like this for them and we were already far along with Endava. Cc
> [[user5]]
>
> [[user10]]

> [!slack]+ Message from User4
> **Time:** 4:04 PM
>
> That works for me, just want to double-confirm that we feel the benefits we gain with Endava outweigh some of the benefits Infosys had mentioned:
> They have 200+ Guidewire-certified employees.
> They already have sandbox instances we'd be able to use for the Connectors without paying to obtain them from Guidewire.
> They take on contracted liability for ongoing support/maintenance/upgrades of marketplace integrations.
> They bring a list of 195 users.
> As long as we're happy with what Endava offers in comparison (e.g., speed to ramp) and the benefits of going with Endava will drive the ultimate goal of expanding our customer pipeline, then I'm peaceful with the decision.
> I think there is value in User1 reviewing our planned approach with Infosys to get their take and our A&C team engaging them in our GTM strategy
> Whichever we're going with, I think we should still have the quick follow-up call with Infosys, since it will not even include their offshore Guidewire development team next week (it will be more information-trading rather than validation/proposal of any solution, which
> [[user1]]
>
> [[user6]]
> I think you still may find intriguing). More than anything, my hope is that we don't lose momentum on the partnership in general, as we already know it's leading us to 3 named user workshops and a list of 195 other Insurance businesses.
> [[user7]]
> let me know if I'm thinking about it the wrong way, but a discussion on the topic of connectors doesn't come with an obligation or an expectation-we can do the quick discovery and then come back and say "let's focus on direct to start because we have the demand". (edited)
>
> :bufo-clap: 1

> [!slack]+ Message from User8
> **Time:** 4:11 PM
>
> One more chime in here - is there a world in which we use both? Does the GW work need to be exclusive? (edited)
>
> User4
>   Yesterday at 4:3 PM
> I'm down to sync in a bit if it's helpful for us all to connect briefly on this topic! Would just like to understand a little more about decisions and next steps and context I might have missed from separate threads, and then happy to disagree and commit if needed!
> 4:34
> One more chime in here - is there a world in which we use both? Does the GW work need to be exclusive?
> For Connectors it's probably best to go with one for the suite, but then we could use other partners like Infosys for the custom Guidewire integrations with users.

> [!slack]+ Message from User4
> **Time:** 4:33 PM
>
> I'm down to sync in a bit if it's helpful for us all to connect briefly on this topic! Would just like to understand a little more about decisions and next steps and context I might have missed from separate threads, and then happy to disagree and commit if needed!
> 4:34
> One more chime in here - is there a world in which we use both? Does the GW work need to be exclusive?
> For Connectors it's probably best to go with one for the suite, but then we could use other partners like Infosys for the custom Guidewire integrations with users.

> [!slack]+ Message from User14
> **Time:** 4:35 PM
>
> Correct
>
> User4
>   Yesterday at 4:37 PM
> Let me know if you're free to connect this afternoon real quick
> [[user6]]
>
> [[user1]]
>
> [[user7]]
>
> [[user8]]
> , feels like we can quickly get on the same page and move forward all with shared context

> [!slack]+ Message from User4
> **Time:** 4:37 PM
>
> Let me know if you're free to connect this afternoon real quick
> [[user6]]
>
> [[user1]]
>
> [[user7]]
>
> [[user8]]
> , feels like we can quickly get on the same page and move forward all with shared context

> [!slack]+ Message from User1
> **Time:** 4:37 PM
>
> Happy to jump on a quick call.

> [!slack]+ Message from User4
> **Time:** 5:50 PM
>
> User1 and I just caught up, here's a recap:
> Most importantly, User1 flagged that the benefit of speed we get from Endava is partially because they're in our LDAP, they're in our JIRA, we have an SoW with them-basically, all the legal/logistical/admin details are already taken care of, they're fully integrated.
> [[user7]]
>
> [[user9]]
> do you know if Infosys is already set up with us in the same way from any prior engagement? If not, we'l have to go through contracting, legal redlines, provisioning, permissions, etc ... this process will push us far past the June 30th integration deadline. If this is the case, the Connector discussion is a moot point and we keep focus on Guidewire-to-user direct.
> If there are no admin roadblocks and Infosys is fully contracted and in our systems, then User1 is open to a quick discussion if they can get it scheduled ASAP next week, just in case they bring something to the table that blows Endava out of the water. We've already stressed in the email thread that we're on a major time crunch to make a decision; we'l be sure to stress this in the meeting as well in order to set clear expectations.
> We will highlight that, Connector or not, we're committed to continuing discovery with them on non-Marketplace Guidewire-to-user integrations. The direct opportunities with Travelers, MassMutual, and The Hartford are still very much something we want to pursue, and we also have no shortage of Guidewire-to-user opportunities that could use the expertise of a dedicated partner if there's a fit.

> [!slack]+ Message from User7
> **Time:** 6:49 PM
>
> Infosys does not have these contractual agreements in place.
> [[user4]]
> let's talk strategy on this on Monday. There's some history here
>
> :ack-sg2: 1"
`;

exports[`Slack Formatter Validation Samples should correctly format multi-person-dm-sample.txt 1`] = `
"> [!slack]+ Message from User1
> **Time:** 3:13 PM
>
> Hello chaps! Happy March!
> 3:13
> Wanted to get your feedback on my update Guidewire doc here https://docs.google.com/document/d/1l0LcYAZxgSNDNCNTmZaOBvVuyTt4fosZu6Lr49PXxWs/edit?tab=t.0
>
> Google Docs
>
> Guidewire-Stripe Connector: Consolidated Strategy & Decision Document

> [!slack]+ Message from Google Doc
> **Time:** 3:14 AM
>
> The part im missing is customer validation - are you guys able to help me out with some customer demand that I can list for these connectors?
>
> User2

> [!slack]+ Message from User2
> **Time:** 3:14 PM
>
> hi
> [[user1]]
> will look at it in a bit.. great to hear from you..

> [!slack]+ Message from User1
> **Time:** 3:14 PM
>
> You too dude! Hope you're not too sick!
> 3:14
> My whole household had Norovirus last week, fun times (edited)
>
> User2

> [!slack]+ Message from User2
> **Time:** 3:15 PM
>
> oh boy.. the work distracts me from thinking about it and makes the day go by faster

> [!slack]+ Message from User1
> **Time:** 3:16 PM
>
> works to a point, noro there's no point even trying to fight it
> 3:16
> just buckle up for the ride LOL
>
> User3
>   3:20 PM
> Cheers gentlemen!
> 3:20
> I will take a look at the doc
> [[user1]]

> [!slack]+ Message from User3
> **Time:** 3:20 PM
>
> Cheers gentlemen!
> 3:20
> I will take a look at the doc
> [[user1]]

> [!slack]+ Message from User1
> **Time:** 3:20 PM
>
> cool thank you

> [!slack]+ Message from User3
> **Time:** 3:20 PM
>
> I am sure it is of the utmost quality
>
> :llm-intensifies: 1

> [!slack]+ Message from User1
> **Time:** 3:21 PM
>
> dont get too hung up on costs those are under negotiation
> 3:21
> it'l be toward the lower end of the range I gave

> [!slack]+ Message from User3
> **Time:** 3:21 PM
>
> yea I never do - that is sales jargon

> [!slack]+ Message from User1
> **Time:** 3:22 PM
>
> feel free to drop comments on it, it's a WIP"
`;
